packages/cli/src/telemetry/index.js

```diff
@@ -66,6 +66,17 @@ export async function startTelemetry() {
     traceProvider.addSpanProcessor(traceProcessor)
     traceProvider.register()
 
+    // If these signals have no listeners, then nodejs will exit without triggering the
+    // process exit event, so we need to ensure we have at least one listener for each
+    for (const signal of ['SIGTERM', 'SIGINT', 'SIGHUP']) {
+      process.on(signal, () => {
+        if (process.listenerCount(signal) === 1) {
+          console.log(`Received ${signal} signal, exiting...`)
+          process.exit()
```

Comment:
I read your PR comment and saw you quoted this from the Node docs
> 'SIGTERM' and 'SIGINT' have default handlers on non-Windows platforms that reset the terminal mode before exiting with code 128 + signal number. If one of these signals has a listener installed, its default behavior will be removed (Node.js will no longer exit).

That, combined with your code comment here leads me to believe:

 * If you **don't** listen to SIGTERM etc node will just exit, without triggering `process.on('exit', () => {})`
 * If yo **do** listen to SIGTERM etc node will not exit at all on SIGTERM etc. That's why you have to manually call `process.exit()`
 
 Is that correct?
 
 If it is correct, could you please add a comment above this `process.exit()` line describing that behavior? Thanks :)

Reply:
Happy to update the comments to clarify this. 

Yeah my understanding is the same. You must register a handler on the interrupts or node will internally exit without firing the "exit event". I have to listen for the interrupts and if no-one else has decided to handle it then I'll simply call process.exit to fire the exit event which will result in the appropriate teardown.